src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaRepository.java

```diff
@@ -77,20 +74,25 @@ public void createTopic(final String topic) {
 
     @Override
     public void createTopic(final String topic, final int partitionsNum, final int replicaFactor,
-                            final long retentionMs, final long rotationMs) {
-
-        final String connectionString = zkFactory.get().getZookeeperClient().getCurrentConnectionString();
-        final ZkUtils zkUtils = ZkUtils.apply(connectionString, settings.getZkSessionTimeoutMs(),
-                settings.getZkConnectionTimeoutMs(), false);
+                            final long retentionMs, final long rotationMs) throws TopicCreationException {
+        ZkUtils zkUtils = null;
         try {
+            final String connectionString = zkFactory.get().getZookeeperClient().getCurrentConnectionString();
```

Comment:
I just wrapped everything in a try/catch in order to be able to treat it properly in the controller. I thought about handling errors in a finer grained style, but I do need to know wether anything wrong happened with this operation in order to undo the persistency.